### DISCUSSION TITLE
Link BasePrice to SalesContractPosition after pricing calculation

### DISCUSSION THREAD

**Discussion by fschon** [opencrx/opencrx#141] [Ideas]
After a sales contract position has been priced, the pricing information from the BasePrice is copied to the SalesContractPosition.
In my case (I am working on incorporating subscription pricing) it would be very useful to have direct access from the SalesContractPosition to the BasePrice used by the pricing rule. Currently I have to look it up each time using:

```
public static ProductBasePrice getProductBasePrice(PriceLevel priceLevel, Product product) {
        return product.getBasePrice().stream()
                .filter(bp -> {
                    return bp.getPriceLevel().stream()
                            .filter(pl -> pl.getIdentity().equals(priceLevel.getIdentity()))
                            .anyMatch(e -> true);
                })
                .findFirst()
                .orElse(null);
}
```

**Comment by wfro64** [comment]
The base price calculated by the pricing rule - more precisely by SalesContractPosition::reprice() - is stored in SalesContractPosition::listPrice. See
* https://www.opencrx.org/opencrx/5.3/uml/opencrx-core/contract1/tn/090_SalesContractPosition.png.html
* https://github.com/opencrx/opencrx/blob/master/core/src/main/java/org/opencrx/kernel/backend/Contracts.java#L1901

**Comment by fschon** [comment]
Brilliant. Thank you Werner.
